### PR TITLE
test calcul dhekra

### PR DESCRIPTION
**Pull request by dhekraessid**

### CODE REVIEW

File: src/main/scala/com/hs/pipeline/demo/schema/Contract.scala [hunk 1]
```diff
@@ -22,7 +24,8 @@ object Contract {
       Amount,
       Country,
       Entity,
-      OrderTime
+      OrderTime,
+    participation
```
Comment (hassaneseoud) [inline src/main/scala/com/hs/pipeline/demo/schema/Contract.scala]: contract raw schema does not include a "participation" column, it's a calculated column, you can define it in the BV schema or somewhere else

File: src/main/scala/com/hs/pipeline/demo/schema/Contract.scala [hunk 2]
```diff
@@ -38,5 +41,8 @@ object Contract {
     selectAndRenameColumns(selectedCols, renameMapping)(df)
   }
 
+  def gitcalculateParticipation (contractDf: DataFrame): DataFrame = {
```
Comment (hassaneseoud) [inline src/main/scala/com/hs/pipeline/demo/schema/Contract.scala:44]: this function is not used

File: src/main/scala/com/hs/pipeline/demo/schema/Contract.scala (hunk 1 above)
Comment (dhekraessid) [reply src/main/scala/com/hs/pipeline/demo/schema/Contract.scala]: en fait BV ça refere à quoi ?

File: src/main/scala/com/hs/pipeline/demo/schema/Contract.scala [hunk 3]
```diff
@@ -41,8 +52,15 @@ object Contract {
     selectAndRenameColumns(selectedCols, renameMapping)(df)
   }
 
-  def gitcalculateParticipation (contractDf: DataFrame): DataFrame = {
+  def calculateParticipation(contractDf: DataFrame): DataFrame = {
+    contractDf.withColumn("sum", sum("Amount").over(Window.partitionBy("Entity")))
```
Comment (hassaneseoud) [inline src/main/scala/com/hs/pipeline/demo/schema/Contract.scala:56]: partitionBy(Entity) as constant "entity_id" and not Entity

File: src/test/ressources/calculateParticipation/result.csv [hunk 4]
```diff
@@ -1,7 +1,7 @@
-id;entity_id;amount
```
Comment (hassaneseoud) [inline src/test/ressources/calculateParticipation/result.csv]: we should respect the same pattern, once we started with files names in uppercase we should keep it

**Pull request closed without merging**